app/services/operations/archive_application.rb

```diff
@@ -0,0 +1,18 @@
+module Operations
+  class ArchiveApplication
+    def initialize(application_id:)
+      @application = CrimeApplication.find(application_id)
+    end
+
+    def call
+      raise Errors::AlreadyArchived if application.archived?
+      raise Errors::CannotArchive unless application.returned?
+
+      application.update!(archived: true, archived_at: Time.zone.now)
```

Comment:
Do we need an archived column? I'd imagined we'd have a consumer_scope like:

`where(archived_at: nil)`

Reply:
You're right, I think `archived_at` is sufficient.